pallets/erc20-xcm-bridge/src/lib.rs

```diff
@@ -85,6 +85,16 @@ pub mod pallet {
 			// append amount to be transferred
 			input.extend_from_slice(H256::from_uint(&amount).as_bytes());
 
+			let weight_limit = pallet_evm::FixedGasWeightMapping::<T>::gas_to_weight(
+				T::Erc20TransferGasLimit::get(),
+				true,
+			);
+
+			let transaction_len = (erc20_contract_address.encode().len()
```

Comment:
No, there is no transaction in the block for this evm transfer, you should provide Some(0)